### PR TITLE
DEVX-1952: New ccloud example: CLI now supports JSON/Avro for consume/produce

### PR DESCRIPTION
**Pull request by ybyzek**
@rspurgeon note the `expect` workaround.

### CODE REVIEW

File: clients/cloud/ccloud/README.md [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+$ ccloud kafka topic produce test2 --value-format avro --schema schema.json
+```
+
+Type a few messages, using a `,` as the separator between the message key and value:
+
+```bash
+{"count":3}
+{"count":4}
+{"count":5}
+```
```
Comment (rspurgeon) [inline clients/cloud/ccloud/README.md]: This asks the users to use a `key,value` event, but doesn't include the `--parse-key --delimiter ,` flags or explanation like the non-avro example does.

File: clients/cloud/ccloud/README.md [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+$ ccloud kafka topic consume test2 -b --value-format avro
+```
+
+You should see the messages you typed in the previous step.
+
+```bash
+{"count":3}
+{"count":4}
+{"count":5}
+```
```
Comment (rspurgeon) [inline clients/cloud/ccloud/README.md]: * The long name flags are double `--`
* Did you intend for the user to use the `--print-key` flag in the example?

File: clients/cloud/ccloud/ccloud-ccsr-example.sh [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+  || exit 1
+ccloud::validate_version_ccloud_cli 1.13.0 \
+  && print_pass "ccloud version ok" \
+  || exit 1
+ccloud::validate_logged_in_ccloud_cli \
+  && print_pass "logged into ccloud CLI" \
+  || exit 1
+
+if [[ -z "$SR_API_KEY" ]] || [[ -z "$SR_API_SECRET" ]] ; then
+  echo "ERROR: You must export SR_API_KEY and SR_API_SECRET before running this script"
```
Comment (rspurgeon) [inline clients/cloud/ccloud/ccloud-ccsr-example.sh]: nit ⚠️ : It's not required that they be "exported", but just that they be "set"
```suggestion
  echo "ERROR: The SR_API_KEY and SR_API_SECRET variables are required to be set when running this script"
```

File: clients/cloud/ccloud/README.md (hunk 2 above)
Comment (ybyzek) [reply clients/cloud/ccloud/README.md]: No key, pulling out `--print-key`

File: clients/cloud/ccloud/README.md (hunk 1 above)
Comment (ybyzek) [reply clients/cloud/ccloud/README.md]: No key, pulling out verbiage about the key

File: clients/cloud/ccloud/README.md [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+alice,{"count":0}
+alice,{"count":1}
+alice,{"count":2}
+```
+
+When you are done, press `<ctrl>-d`.
+
+2. Run the [Confluent Confluent CLI consumer](https://docs.confluent.io/current/cloud/cli/command-reference/kafka/topic/ccloud_kafka_topic_produce.html#ccloud-kafka-topic-produce?utm_source=github&utm_medium=demo&utm_campaign=ch.examples_type.community_content.clients-ccloud), reading messages from topic `test1`, passing in additional arguments:
+
+* `-print-key`: print key and value (by default, it only prints value)
```
Comment (rspurgeon) [inline clients/cloud/ccloud/README.md]:
```suggestion
* `--print-key`: print key and value (by default, it only prints value)
```

File: clients/cloud/ccloud/README.md [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+Type a few messages, using a `,` as the separator between the message key and value:
+
+```bash
+alice,{"count":0}
+alice,{"count":1}
+alice,{"count":2}
+```
+
+When you are done, press `<ctrl>-d`.
```
Comment (rspurgeon) [inline clients/cloud/ccloud/README.md:40]: Sometimes for exiting, we instruct `<ctrl>-c` and other times we use `<ctrl>-d`.  IMO we stick with `<ctrl>-c`

File: clients/cloud/ccloud/README.md (hunk 2 above)
Comment (ybyzek) [reply clients/cloud/ccloud/README.md]: @rspurgeon reversal: going to fix the examples to use the key

File: clients/cloud/ccloud/README.md (hunk 4 above)
Comment (ybyzek) [reply clients/cloud/ccloud/README.md]: Fixed in commit

File: clients/cloud/ccloud/ccloud-ccsr-example.sh (hunk 3 above)
Comment (ybyzek) [reply clients/cloud/ccloud/ccloud-ccsr-example.sh]: Addressed in separate commit

File: clients/cloud/ccloud/ccloud-ccsr-example.sh [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+  echo "ERROR: The variables SR_API_KEY and SR_API_SECRET must be set before running this script ('export SR_API_KEY=<>', 'export SR_API_SECRET=<>')"
+  exit 1
+fi
+
+# Set topic name
+topic_name=test2
+
+# Create topic in Confluent Cloud
+echo -e "\n# Create topic $topic_name"
+ccloud kafka topic create $topic_name || true
```
Comment (rspurgeon) [inline clients/cloud/ccloud/ccloud-ccsr-example.sh]: Does the `|| true` do something here?

Comment (ybyzek) [reply clients/cloud/ccloud/ccloud-ccsr-example.sh]: No but...this leads to the discovery that the CLI now supports `--if-not-exists` flag ;)

Comment (rspurgeon) [reply clients/cloud/ccloud/ccloud-ccsr-example.sh]: 💯 💯 💯

**Review by rspurgeon — APPROVED** [review]
LGTM, nice addition

**Pull request merged**